lib/Rex/Commands/Partition.pm

```diff
@@ -225,42 +226,8 @@ sub partition {
     sleep 1;
   }
 
-  if ( !exists $option{fstype}
-    || $option{fstype} eq "non-fs"
-    || $option{fstype} eq "none"
-    || $option{fstype} eq "" )
-  {
-    # nix
-  }
-  elsif ( can_run("mkfs.$option{fstype}") ) {
-    Rex::Logger::info(
-      "Creating filesystem $option{fstype} on /dev/$disk$part_num");
-
-    my $add_opts = "";
-
-    if ( exists $option{label} || exists $option{lable} ) {
-      my $label = $option{label} || $option{lable};
-      $add_opts .= " -L $label ";
-    }
-
-    run "mkfs.$option{fstype} $add_opts /dev/$disk$part_num";
-  }
-  elsif ( $option{fstype} eq "swap" ) {
-    Rex::Logger::info("Creating swap space on /dev/$disk$part_num");
-
-    my $add_opts = "";
-
-    if ( exists $option{label} || exists $option{lable} ) {
-      my $label = $option{label} || $option{lable};
-      $add_opts .= " -L $label ";
-    }
-
-    run "mkswap $add_opts /dev/$disk$part_num";
-  }
-  else {
-    die("Can't format partition with $option{fstype}");
-  }
-
+  mkfs "$disk$part_num";
```

Comment:
This PR looks like was merged for some reason without resolving this part of the problem, and lead us to #905 :(